tests/test_helpers/dataset_utils.py

```diff
@@ -0,0 +1,258 @@
+            field["fields"] = generate_fields(value[0], current_path, field_map)
+        else:
+            if existing_field := get_existing_field(field_map, current_path):
+                if isinstance(existing_field, ScalarField):
+                    # field exists, copy existing data categories and data_type (if available)
+                    field["data_categories"] = existing_field.data_categories or [
+                        "system.operations"
+                    ]
+                    data_type = (
+                        existing_field.data_type()
+                        if existing_field.data_type() != "None"
+                        else data_type
+                    )
+                    if data_type:
+                        field["fidesops_meta"] = {"data_type": data_type}
+                elif isinstance(existing_field, ObjectField):
+                    # the existing field has a more complex type than what we could derive
+                    # from the API response, we need to copy the fields too instead of just
+                    # the data_categories and data_type
+                    field["fidesops_meta"] = {
+                        "data_type": "object[]" if isinstance(value, list) else "object"
+                    }
+                    field["fields"] = get_simple_fields(existing_field.fields.values())
+            else:
+                # we don't have this field in our dataset, use the default category
+                # and the derived data_type
+                field["data_categories"] = ["system.operations"]
+                # we don't assume the data_type for empty strings, empty lists,
+                # empty dicts, or nulls
+                if data_type:
+                    field["fidesops_meta"] = {"data_type": data_type}
+        fields.append(field)
+    return fields
+
+
+def get_existing_field(field_map: Dict[str, Collection], path: str) -> Optional[Field]:
+    """
+    Lookup existing field by collection name and field path.
+    """
+    collection_name, field_path = path.split(".", 1)
+    if collection := field_map.get(collection_name):
+        return collection.field_dict.get(FieldPath.parse((field_path)))
+    return None
+
+
+def get_simple_fields(fields: Iterable[Field]) -> List[Dict[str, Any]]:
+    """
+    Converts dataset fields into simple dictionaries with only
+    name, data_category, and data_type.
+    """
+
+    object_list = []
+    for field in fields:
+        object = {"name": field.name}
+        if field.data_categories:
+            object["data_categories"] = field.data_categories
+        if field.data_type() != "None":
+            object["fidesops_meta"] = {"data_type": field.data_type()}
+        if isinstance(field, ObjectField) and field.fields:
+            object["fields"] = get_simple_fields(field.fields.values())
+        object_list.append(object)
+    return object_list
+
+
+def get_data_type(value) -> Optional[str]:
```

Comment:
to me, it feels like this function should live more centrally. what would you think of defining it somewhere within `fidesops.graph.data_type.py`? from what i can tell, there's nothing yet defined there that takes an arbitrary input and gives the corresponding proper `Fidesops` `DataType`, but I think that's what you want here.

If you add that utility method over there, then I think here you should be relying upon the proper `DataType` enum - passing around string literals here just feels a bit sloppy to me. Specifically, I think it's ripe for unnoticed regressions if/when any of the typing logic changes in the fidesops "taxonomy" (for lack of a better term), and it will be burdensome to maintain.